plugin-build/plugin/src/main/java/ru/kode/android/build/publish/plugin/BuildPublishPlugin.kt

```diff
@@ -131,13 +133,39 @@ abstract class BuildPublishPlugin : Plugin<Project> {
     }
 }
 
-private fun prepareBuildVariants(ext: ApplicationExtension): Set<String> {
-    return if (ext.productFlavors.isEmpty()) {
-        ext.buildTypes.map { it.name }.toSet()
-    } else {
-        ext.productFlavors.flatMap { flavor ->
-            ext.buildTypes.map { "${flavor.name}${it.name.capitalize()}" }
-        }.toSet()
+private fun ApplicationExtension.extractBuildVariants(): Set<String> {
```

Comment:
I found a case where flavor and buildType name should be separately. For example path to apk file has relative path "/flavor/buildTypeName". Can we change concatenated str to record:
```
data class BuildVariant(
   val flavor: String,
   val buildType: String,
   val dimension: String,
) 
...
fun BuildVariant.format(): String {
    ... 
}
```
??

Reply:
Yep. But I think it is more correct to implement data structure like that:
```
data class BuildVariant(
   val flavorNames: List<String>,  // list of flavors ordered by dimension priority
   val buildTypeName: String,
) 

fun BuildVariant.format(): String { ... }
```
Cause in case with multiple dimensions you will get more complex apk path: "flavorNameFromDim1/flavorNameFromDim2/.../flavorNameFromDimN/buildTypeName/*.apk".